### PR TITLE
[Feature] Support 2S-AGCN

### PR DESCRIPTION
**Pull request by gengenkai**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily got feedback.
If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduces changes that break the back-compatibility of this repo?
If so, please describe how it breaks the compatibility and how users should modify their codes to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools should be used to fix the potential lint issues.
2. The modification should be covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation should be modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmaction/models/backbones/agcn_2s.py [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+from mmcv.cnn import constant_init, kaiming_init, normal_init
+from mmcv.runner import load_checkpoint
+from mmcv.utils import _BatchNorm
+
+from ...utils import get_root_logger
+from ..builder import BACKBONES
+from ..skeleton_gcn.utils import Graph
+
+
+def conv_branch_init(conv, branches):
```
Comment (kennymckormick) [inline mmaction/models/backbones/agcn_2s.py:14]: Can we use init functions in mmcv?

File: mmaction/models/backbones/agcn_2s.py [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+            z = self.conv_d[i](torch.matmul(A2, A1).view(N, C, T, V))
+            y = z + y if y is not None else z
+        y = self.bn(y)
+        y += self.down(x)
+
+        return self.relu(y), adj_mat
+
+
+@BACKBONES.register_module()
+class AGCN_2S(nn.Module):
```
Comment (kennymckormick) [inline mmaction/models/backbones/agcn_2s.py:232]: You can call this model AGCN. 2s mean 2stream, which means two-stream late fusion, not related to the AGCN model.

File: tools/data/skeleton/gen_ntu_rgbd_bone.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import os
```
Comment (kennymckormick) [inline tools/data/skeleton/gen_ntu_rgbd_bone.py:1]: Why do we need another script for bone feature extraction? This can not be conducted by `gen_ntu_rgbd_raw.py`?

File: configs/skeleton/2s-agcn/2sagcn_80e_ntu60_xsub_keypoint_3d.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        ann_file=ann_file_val,
+        data_prefix='',
+        pipeline=test_pipeline))
+
+# optimizer
+optimizer = dict(
+    type='SGD', lr=0.1, momentum=0.9, weight_decay=0.0001, nesterov=True)
+optimizer_config = dict(grad_clip=None)
+# learning policy
+lr_config = dict(policy='step', step=[30, 40])
```
Comment (kennymckormick) [inline configs/skeleton/2s-agcn/2sagcn_80e_ntu60_xsub_keypoint_3d.py:64]: Are you sure about the step? 30, 40 means you perform lr_decay at the 30th and 40th epoch ...

**Review by kennymckormick — COMMENTED** [review]
You also need to add missing documentations.

File: mmaction/datasets/pipelines/formatting.py [hunk 5]
```diff
@@ -374,6 +374,64 @@ def __repr__(self):
...
+    Required keys are "keypoint" ,
+    added or modified keys are "keypoint".
+
+    Args:
+        dataset (str): Define the type of dataset.
+    """
+
+    def __init__(self, dataset='xsub'):
+        self.dataset = dataset
+        if self.dataset not in ['xview', 'xsub', 'kinectics']:
```
Comment (kennymckormick) [inline mmaction/datasets/pipelines/formatting.py:390]: 2 types indeed: NTURGB+D (25p), OpenPose (18p). Maybe you can also add COCO.

File: mmaction/datasets/pipelines/formatting.py [hunk 6]
```diff
@@ -374,6 +374,64 @@ def __repr__(self):
...
+            raise ValueError(
+                f'The dataset type {self.dataset} is not supported')
+        if self.dataset == 'xview':
```
Comment (kennymckormick) [inline mmaction/datasets/pipelines/formatting.py:393]: 'xview' and 'xset' are the same, need to be merged.

File: mmaction/datasets/pipelines/formatting.py [hunk 7]
```diff
@@ -374,6 +374,64 @@ def __repr__(self):
...
+
+        Args:
+            results (dict): The resulting dict to be modified and passed
+                to the next transform in pipeline.
+        """
+        keypoint = results['keypoint']
+        M, T, V, C = keypoint.shape
+        bone = np.zeros((M, T, V, C), dtype=np.float32)
+
+        for v1, v2 in self.pairs:
```
Comment (kennymckormick) [inline mmaction/datasets/pipelines/formatting.py:422]: u serious? Note that for kinetics, there is a pair (0, 0), then it will be (-1, -1)

File: mmaction/datasets/pipelines/formatting.py [hunk 8]
```diff
@@ -374,6 +374,64 @@ def __repr__(self):
...
+            v1 -= 1
+            v2 -= 1
+            bone[:, :, v1, :] = keypoint[:, :, v1, :] - keypoint[:, :, v2, :]
```
Comment (kennymckormick) [inline mmaction/datasets/pipelines/formatting.py:425]: What about the score? For OpenPose KP, the 3rd channel in the last dim is score.

File: mmaction/datasets/pipelines/formatting.py [hunk 9]
```diff
@@ -374,6 +374,64 @@ def __repr__(self):
...
 
+@PIPELINES.register_module()
+class JointToBone:
+    """Convert the joint information to bone information.
+
+    Required keys are "keypoint" ,
+    added or modified keys are "keypoint".
+
+    Args:
+        dataset (str): Define the type of dataset.
```
Comment (kennymckormick) [inline mmaction/datasets/pipelines/formatting.py:385]: docstring. need to specify the default value and the applicable choices.

**Comment by kennymckormick** [comment]
Besides, the related documents are missing, including README (mimick the format of the **latest** README files). README_zh-CN, metafile (and changelogs).
